File: sdk/src/main/java/com/kin/ecosystem/main/presenter/IEcosystemPresenter.java
package com.kin.ecosystem.main.presenter;

import android.os.Bundle;
import com.kin.ecosystem.base.IBasePresenter;
import com.kin.ecosystem.main.ScreenId;
import com.kin.ecosystem.main.view.IEcosystemView;

public interface IEcosystemPresenter extends IBasePresenter<IEcosystemView> {

	void onStart();

	void onStop();

	void balanceItemClicked();

	void backButtonPressed();

	void visibleScreen(@ScreenId final int id);

	void settingsMenuClicked();

	void onMenuInitialized();


	void onSaveInstanceState(Bundle outState);
}

File: sdk/src/main/java/com/kin/ecosystem/main/presenter/IEcosystemPresenter.kt
package com.kin.ecosystem.main.presenter

import android.os.Bundle
import com.kin.ecosystem.base.IBasePresenter
import com.kin.ecosystem.main.ScreenId
import com.kin.ecosystem.main.view.IEcosystemView

interface IEcosystemPresenter : IBasePresenter<IEcosystemView> {

    fun onStart()
    
    fun touchedOutside()

    fun backButtonPressed()

    fun visibleScreen(id: ScreenId)

    fun onSaveInstanceState(outState: Bundle)
}
